test/jdk/tools/jlink/plugins/SaveJlinkArgfilesPluginTest.java

```diff
@@ -0,0 +1,117 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Test --save-jlink-argfiles plugin
+ * @library ../../lib
+ * @library /test/lib
+ * @modules java.base/jdk.internal.jimage
+ *          jdk.jdeps/com.sun.tools.classfile
+ *          jdk.jlink/jdk.tools.jlink.internal
+ *          jdk.jlink/jdk.tools.jmod
+ *          jdk.jlink/jdk.tools.jimage
+ *          jdk.compiler
+ * @build tests.*
+ * @run main SaveJlinkArgfilesPluginTest
+ */
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import jdk.test.lib.process.ProcessTools;
+
+import tests.Helper;
+
+public class SaveJlinkArgfilesPluginTest {
+
+    public static void main(String[] args) throws Throwable {
+
+        Helper helper = Helper.newHelper();
+        if (helper == null) {
+            System.err.println("Test not run");
+            return;
+        }
+
+        String exe = System.getProperty("os.name").startsWith("Windows") ? ".exe" : "";
+        Path argfile1 = Path.of("argfile1");
+        Path argfile2 = Path.of("argfile2");
+
+        Files.writeString(argfile1, "--add-modules jdk.internal.vm.ci --add-options=-Dfoo=xyzzy");
+        Files.writeString(argfile2, "--vendor-version=\"XyzzyVM 3.14.15\" --vendor-bug-url=https://bugs.xyzzy.com/");
+
+        var module = "base";
+        helper.generateDefaultJModule(module);
+        var image = helper.generateDefaultImage(new String[] {
+                "--add-modules", "jdk.jlink,jdk.jdeps,jdk.internal.opt,jdk.compiler,java.compiler,jdk.zipfs,jdk.internal.vm.ci",
+                "--keep-packaged-modules", "images/base.image/jmods",
+                "--save-jlink-argfiles", argfile1 + File.pathSeparator + argfile2
+            }, module)
+            .assertSuccess();
+        helper.checkImage(image, module, null, null);
+
+        Path launcher = image.resolve("bin/java" + exe);
+        var oa = ProcessTools.executeProcess(launcher.toString(), "-XshowSettings:properties", "--version");
+
+        // Check that the primary image creation ignored the saved args
+        oa.shouldHaveExitValue(0);
+        oa.shouldNotMatch("yzzy");
```

Comment:
Can this check for `java.vendor.version = XyzzyVM 3.14.15`, that will also help the reader to understand what the test tries to verify.

Reply:
Not sure follow as this is a negative match.